client/state/themes/test/reducer.js

```diff
@@ -402,4 +408,220 @@ describe( 'reducer', () => {
+				meta: {
+					analytics: [
+						{
+							payload: {
+								name: 'calypso_themeshowcase_theme_activate',
+								properties: {
+									previous_theme: 'twentyfifteen',
+									purchased: false,
+									search_term: 'simple, white',
+									source: 'unknown',
+									theme: 'twentysixteen',
+								},
+								service: 'tracks',
+							},
+							type: 'ANALYTICS_EVENT_RECORD'
+						},
+					],
+				},
+				type: THEME_ACTIVATE_REQUEST_SUCCESS,
+				theme: { id: 'twentysixteen' },
+				siteId: 2211888,
+			} );
+
+			expect( state ).to.have.keys( [ '2211888' ] );
+			expect( state ).to.eql( { 2211888: 'twentysixteen' } );
+		} );
+
+		it( 'should persist state', () => {
+			const original = deepFreeze( activeThemes( deepFreeze( {} ), {
+				meta: {
+					analytics: [
+						{
+							payload: {
+								name: 'calypso_themeshowcase_theme_activate',
+								properties: {
+									previous_theme: 'twentyfifteen',
+									purchased: false,
+									search_term: 'simple, white',
+									source: 'unknown',
+									theme: 'twentysixteen',
+								},
+								service: 'tracks',
+							},
+							type: 'ANALYTICS_EVENT_RECORD'
+						},
+					],
+				},
+				type: THEME_ACTIVATE_REQUEST_SUCCESS,
+				theme: { id: 'twentysixteen' },
+				siteId: 2211888,
+			} ) );
+
+			const state = activeThemes( original, { type: SERIALIZE } );
+
+			expect( state ).to.eql( { 2211888: 'twentysixteen' } );
+		} );
+
+		it( 'should load valid persisted state', () => {
+			const original = deepFreeze( { 2211888: 'twentysixteen' } );
+
+			const state = activeThemes( original, { type: DESERIALIZE } );
+			expect( state ).to.eql( { 2211888: 'twentysixteen' } );
+		} );
+
+		it( 'should not load invalid persisted state', () => {
+			const original = deepFreeze( {
+				2916284: 1234
+			} );
+
+			const state = activeThemes( original, { type: DESERIALIZE } );
+			expect( state ).to.eql( {} );
+		} );
+	} );
+
+	describe( 'activeThemeRequest', () => {
+		it( 'should default to an empty object', () => {
+			const state = activeThemeRequest( undefined, {} );
+			expect( state ).to.eql( {} );
+		} );
+
+		it( 'should map site ID to true value if request in progress', () => {
+			const state = activeThemeRequest( deepFreeze( {} ), {
+				type: ACTIVE_THEME_REQUEST,
+				siteId: 2916284,
+			} );
+
+			expect( state ).to.eql( {
+				2916284: true
+			} );
+		} );
+
+		it( 'should accumulate mappings', () => {
+			const state = activeThemeRequest( deepFreeze( {
+				2916284: true
+			} ),
+				{
+					type: ACTIVE_THEME_REQUEST,
+					siteId: 2916285,
+				} );
+
+			expect( state ).to.eql( {
+				2916284: true,
+				2916285: true,
+			} );
+		} );
+
+		it( 'should map site ID to false value if request finishes successfully', () => {
+			const state = activeThemeRequest( deepFreeze( {
+				2916284: true
+			} ), {
+				type: ACTIVE_THEME_REQUEST_SUCCESS,
+				siteId: 2916284,
+				themeId: 'twentysixteen',
+				themeName: 'Twentysixteen',
```

Comment:
I'm thinking we should maybe omit `themeName` and `themeCost` (which aren't used by these reducers anyway, much like the analytics `meta`) since we might not pass those on as part of the action in the future, so I think it'll save us some work to remove them from the tests already. (Even if we keep them, I think they're more distracting here. If we decide to keep and use them for a different reducer, we'll cover them with that reducer's tests.)